applications/vanilla/views/modules/promotedcontent.php

```diff
@@ -1,33 +1,36 @@
 <?php if (!defined('APPLICATION')) exit();
 require_once Gdn::controller()->fetchViewLocation('helper_functions', 'modules', 'vanilla');
+$Content = $this->data('Content');
 
-?>
-<div class="Box BoxPromoted">
-    <?php echo panelHeading(t('Promoted Content')); ?>
-    <div class="PanelInfo DataList">
-        <?php
-        $Content = $this->data('Content');
-        $ContentItems = sizeof($Content);
-
-        if ($Content):
+$ShowIfNoResults = $this->data('ShowIfNoResults');
 
-            if ($this->Group):
-                $Content = array_chunk($Content, $this->Group);
-            endif;
+$ContentItems = is_array($Content) ? sizeof($Content) : 0;
 
-            foreach ($Content as $ContentChunk):
+if ($ContentItems != 0 || $ShowIfNoResults):
```

Comment:
You can condense this down to:

```php
if (!empty($Content) || $this->ShowIfNoResults):
```

`empty` will return `true` for an empty array. `ShowIfNoResults` is already a property on `$this`. You don't need to set `$ContentItems` or `$ShowIfNoResults`.